mantidimaging/gui/test/gui_system_spectrum_test.py

```diff
@@ -56,3 +57,71 @@ def test_add_roi(self) -> None:
             self.assertEqual(final_roi_count, initial_roi_count + 1)
             self.assertIn(f'roi_{i}', self.spectrum_window.roi_table_model.roi_names())
             self.assertIn(f'roi_{i}', self.spectrum_window.spectrum_widget.roi_dict)
+
+    def test_remove_roi(self) -> None:
+        QTest.mouseClick(self.spectrum_window.addBtn, Qt.MouseButton.LeftButton)
+        QTest.qWait(SHORT_DELAY)
+        initial_roi_count = self.spectrum_window.roi_table_model.rowCount()
+
+        QTest.mouseClick(self.spectrum_window.removeBtn, Qt.MouseButton.LeftButton)
+        QTest.qWait(SHORT_DELAY)
+        final_roi_count = self.spectrum_window.roi_table_model.rowCount()
+
+        self.assertEqual(final_roi_count, initial_roi_count - 1)
+        self.assertNotIn(f'roi_{initial_roi_count - 1}', self.spectrum_window.roi_table_model.roi_names())
+        self.assertNotIn(f'roi_{initial_roi_count - 1}', self.spectrum_window.spectrum_widget.roi_dict)
+
+    def test_change_roi_color(self) -> None:
+        QTest.mouseClick(self.spectrum_window.addBtn, Qt.MouseButton.LeftButton)
+        QTest.qWait(SHORT_DELAY)
+
+        new_color = (255, 0, 0, 255)
+        self.spectrum_window.spectrum_widget.change_roi_colour('roi_1', new_color)
+
+        self.assertEqual(self.spectrum_window.spectrum_widget.roi_dict['roi_1'].colour, new_color)
+
+    def test_rename_roi(self) -> None:
+        QTest.mouseClick(self.spectrum_window.addBtn, Qt.MouseButton.LeftButton)
+        QTest.qWait(SHORT_DELAY)
+
+        old_name = 'roi_1'
+        new_name = 'roi_renamed'
+        self.spectrum_window.presenter.rename_roi(old_name, new_name)
+
+        self.assertNotIn(old_name, self.spectrum_window.spectrum_widget.roi_dict)
+        self.assertIn(new_name, self.spectrum_window.spectrum_widget.roi_dict)
+
+    def test_adjust_roi(self):
+        QTest.mouseClick(self.spectrum_window.addBtn, Qt.MouseButton.LeftButton)
+        QTest.qWait(SHORT_DELAY)
+
+        roi_names = self.spectrum_window.presenter.get_roi_names()
+        roi_name = next(name for name in roi_names if name != 'all')
+
+        roi_widget = self.spectrum_window.spectrum_widget.roi_dict[roi_name]
+
+        roi_widget.setPos((roi_widget.pos().x() + 10, roi_widget.pos().y() + 10))
+        self.spectrum_window.presenter.handle_roi_moved()
```

Comment:
It might be possible to do this more realistically by moving the handle. Maybe something like
```
roi_widget.movePoint(0, (10, 20))
QTest.qWait(SHORT_DELAY)

updated_roi = self.spectrum_window.presenter.model.get_roi(roi_name)

self.assertEqual(updated_roi.left, 10)
self.assertEqual(updated_roi.top, 20)
```
That should trigger the signals for everything to get updated. You might need to check which handle number to put as the first arg to `movePoint`